tests/test_smoke.py

```diff
@@ -3012,108 +3039,89 @@ def test_skyserve_spot_recovery():
         [
             f'sky serve up -n {name} -y tests/skyserve/spot/recovery.yaml',
             _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=1),
-            f'{_get_serve_endpoint(name)}; curl -L http://$endpoint | grep "Hi, SkyPilot here"',
+            f'{_SERVE_ENDPOINT_WAIT.format(name=name)}; '
+            'curl -L http://$endpoint | grep "Hi, SkyPilot here"',
             _terminate_gcp_replica(name, zone, 1),
             _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=1),
-            f'{_get_serve_endpoint(name)}; curl -L http://$endpoint | grep "Hi, SkyPilot here"',
+            f'{_SERVE_ENDPOINT_WAIT.format(name=name)}; '
+            'curl -L http://$endpoint | grep "Hi, SkyPilot here"',
         ],
         _TEARDOWN_SERVICE.format(name=name),
         timeout=20 * 60,
     )
     run_one_test(test)
 
 
-@pytest.mark.gcp
-@pytest.mark.sky_serve
-def test_skyserve_base_ondemand_fallback():
+@pytest.mark.serve
+def test_skyserve_base_ondemand_fallback(generic_cloud: str):
     name = _get_service_name()
     test = Test(
         f'test-skyserve-base-ondemand-fallback',
         [
-            f'sky serve up -n {name} -y tests/skyserve/spot/base_ondemand_fallback.yaml',
+            f'sky serve up -n {name} --cloud {generic_cloud} -y tests/skyserve/spot/base_ondemand_fallback.yaml',
             _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=2),
-            f'output=$(sky serve status {name});'
-            'echo "$output" | grep "GCP(\[Spot\]vCPU=2)" && '
-            'echo "$output" | grep "GCP(vCPU=2)";'
+            f'{_SERVE_STATUS_WAIT.format(name=name)}; '
+            'echo "$s" | grep "GCP(\[Spot\]vCPU=2)" && '
+            'echo "$s" | grep "GCP(vCPU=2)";'
```

Comment:
those are not true if we move to generic cloud? we should use cloud repr here iiuc.

Reply:
also should we use `_check_replica_in_status` here

Reply:
Good catch! Changed. Thanks!